week2/solutions/intro_to_property_packages-solution.ipynb

```diff
@@ -0,0 +1,929 @@
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0dc3a487",
+   "metadata": {},
+   "source": [
+    "### Solve for other variables using a specified osmotic pressure\n",
+    "\n",
+    "Rather than fixing the state variables to determine other variables, we can also do the inverse. For example, here we determine the concentration for a given osmotic pressure."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "65ac1a6d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Touch conc_mass_phase_comp\n",
+    "m.fs.feed.properties[0].conc_mass_phase_comp\n",
+    "\n",
+    "# Fix osmotic pressure to desired value\n",
+    "osmotic = 12 * pyunits.bar\n",
+    "m.fs.feed.properties[0].pressure_osm_phase[\"Liq\"].fix(osmotic)\n",
+    "\n",
+    "# Unfix the mass flow of TDS\n",
+    "m.fs.feed.properties[0].flow_mass_phase_comp[\"Liq\", \"TDS\"].unfix()\n",
+    "print(f\"Degrees of freedom: {degrees_of_freedom(m)}\\n\")\n",
+    "\n",
+    "results = solver.solve(m)\n",
+    "print(f\"Solve termination {results.solver.termination_condition}\")\n",
+    "print(\n",
+    "    f\"TDS Concentration: {value(m.fs.feed.properties[0].conc_mass_phase_comp['Liq', 'TDS']):.4f} g/L\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52d378b3",
+   "metadata": {},
+   "source": [
+    "# Try It Yourself #1 \n",
+    "\n",
+    "### Create a feed block with the seawater property package to estimate the osmotic pressure and salinity for the following conditions:\n",
+    "\n",
+    "- Mass flow water = 3.3 kg/s\n",
+    "- Mass flow TDS = 0.55 kg/s\n",
+    "- Temperature = 20 C\n",
+    "- Pressure = 1 bar\n",
+    "\n",
+    "The table of available properties is provided here for reference\n",
+    "\n",
+    "<center><img src=\"../graphics/sw_properties.png\" width=\"600\" /></center>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e0c167ec",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ConcreteModel()\n",
+    "m.fs = FlowsheetBlock(dynamic=False)\n",
+    "m.fs.properties = SeawaterParameterBlock()\n",
+    "\n",
+    "# Continue Flowsheet Setup Below\n",
+    "# ================================\n",
+    "\n",
+    "# Create feed block with seawater parameter block\n",
+    "m.fs.feed = Feed(property_package=m.fs.properties)\n",
+    "# Touch the necessary variables to create them in the model\n",
+    "m.fs.feed.properties[0].pressure_osm_phase\n",
+    "m.fs.feed.properties[0].conc_mass_phase_comp\n",
+    "\n",
+    "# Fix the state variables: mass flow of water, mass flow of TDS, temperature, pressure\n",
+    "m.fs.feed.properties[0].flow_mass_phase_comp[\"Liq\", \"H2O\"].fix(3.3)\n",
+    "m.fs.feed.properties[0].flow_mass_phase_comp[\"Liq\", \"TDS\"].fix(0.55)\n",
+    "m.fs.feed.properties[0].temperature.fix(273 + 20)\n",
+    "m.fs.feed.properties[0].pressure.fix(1e5)\n",
```

Comment:
Add in-line comments about the units and maybe add a high-level note that default units are in SI units. I know we've mentioned it verbally a few times, but I'm not sure if we ever put it in writing here or in the lecture